### DISCUSSION TITLE
Inconsistency of configuration folder between Linux app and docker container?

### DISCUSSION THREAD

**Discussion by everwisher** [abraunegg/onedrive#1995] [Q&A] [answered]
In _[Advanced Configuration of the OneDrive Free Client](https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md)_, the unique configuration folder for each onedrive client configuration is located in `~/.config/my-new-config/`; however in _[Docker](https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md)_, the configuration file is located in `/onedrive/conf`. 

If I'm gonna to run multiple SharePoint site configurations in one Docker container, where should I store the config files?

**Comment by abraunegg** [comment, marked as answer]
@everwisher 
> If I'm gonna to run multiple SharePoint site configurations in one Docker container, where should I store the config files?

Dont take this approach.

1 Docker Container = 1 OneDrive Configuration

In your situation you want 1 Docker Container per SharePoint configuration.

**Comment by everwisher** [comment]
Even one configuration per container, I think the mapped drive is different between running on Linux (~/.config/onedrive) and on Docker(/onedrive/conf/)
________________________________

**Reply by abraunegg** [reply]
When creating Docker container you create 2 Docker volumes - one for configuration, one for your data.

For each instance of a Docker container that you are running you will need *unique* Docker volumes:
* One for the OneDrive data 
* One for your Docker configuration - which is an internal Docker volume

**Example:**
```
docker volume create onedrive_conf_sharepoint_site1
docker volume create onedrive_conf_sharepoint_site2
docker volume create onedrive_conf_sharepoint_site3
...
docker volume create onedrive_conf_sharepoint_site50
```

Then, when running the Docker containers:
```
docker run -it --name onedrive -v onedrive_conf_sharepoint_site1:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" -e "ONEDRIVE_UID=${ONEDRIVE_UID}" -e "ONEDRIVE_GID=${ONEDRIVE_GID}" driveone/onedrive:latest
docker run -it --name onedrive -v onedrive_conf_sharepoint_site2:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" -e "ONEDRIVE_UID=${ONEDRIVE_UID}" -e "ONEDRIVE_GID=${ONEDRIVE_GID}" driveone/onedrive:latest
docker run -it --name onedrive -v onedrive_conf_sharepoint_site3:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" -e "ONEDRIVE_UID=${ONEDRIVE_UID}" -e "ONEDRIVE_GID=${ONEDRIVE_GID}" driveone/onedrive:latest
...
docker run -it --name onedrive -v onedrive_conf_sharepoint_site50:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" -e "ONEDRIVE_UID=${ONEDRIVE_UID}" -e "ONEDRIVE_GID=${ONEDRIVE_GID}" driveone/onedrive:latest
```

**Comment by abraunegg** [comment]
@everwisher 
Documentation has been updated so that instructions for running multiple Docker containers is clear - https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-to-use-multiple-onedrive-accounts--configurations-using-docker
